printable-recipe-generator/src/main.ts

```diff
@@ -0,0 +1,42 @@
+import fs from 'fs';
+import * as process from 'node:process';
+import path from 'path';
+import type { Data } from 'ejs';
+import { render as renderTemplate } from 'ejs';
+
+function renderJsonToHtml(recipeDataPath: string) {
+	//load recipe JSON
+	const recipe = JSON.parse(
+		fs.readFileSync(recipeDataPath, 'utf-8'),
+	) as unknown;
+
+	//load template
+	const templatePath = path.join(__dirname, 'src', 'assets', 'recipe.ejs');
+	const template = fs.readFileSync(templatePath, 'utf-8');
+
+	//Render html
+	let renderHtml = '';
+	try {
+		renderHtml = renderTemplate(template, recipe as Data);
+	} catch (error) {
```

Comment:
with my earlier comment about refactoring, I wouldn't `try/catch` here - I'd do it at the outer level and let `renderJsonToHtml` throw as that is more test-friendly